src/utilities/SimulationBaseClass.py

```diff
@@ -374,6 +374,13 @@ def CreateNewPythonProcess(self, procName, priority = -1):
         :param priority (int): Priority that determines when the model gets updated. (Higher number = Higher priority)
         :return: simulationArchTypes.PythonProcessClass object
         """
+        self.bskLogger.bskLog(
```

Comment:
:small_red_triangle_down: It's not appropriate to use bskLogger for this purpose. The Warnings packages and others provide appropriate deprecation warning integration and control. I'm happy to help you fix this.

I realize that bskLogger maybe used in a similar manner in other places. It should not be and we will eventually refactor that usage also 👍 

https://docs.python.org/3/library/warnings.html

Reply:
@schaubh and I briefly discussed how to implement the deprecation warning and this is the solution we came up with. I am happy to switch to the warnings module. However, we should think about a more permanent non-python-specific way to raise these kinds of warnings.

Reply:
Can I ask why we would want a non python specific mechanism?

Reply:
I agree with Patrick here.  We are doing a depreciation warning, not a usual BSK warning of something being being setup properly.  Let's go with the python based depreciation warning as he recommends.

Reply:
I would prefer a non-python deprecation warning system because it gives consistency between calls in C++ and Python. If we need to deprecate something on the Python and C++ layers, and each layer uses a different system (warnings package vs. bskLogger, for example), end users will see these warnings differently and will have to supress them in different ways. I would rather avoid this lack of consistency.

That being said, I agree that the Python layer is more intuitive and feature-rich and should be prefered.

There are ways for the C++ layer to call Python logger, raise Python Warnings, and even raise Python exceptions, which involves SWIG director feature. However, this would make the logger slower. Should I look into this?

Reply:
I think we might be talking past each other here a bit 😄 . I'm referring specifically of deprecations scoped to the python (non-swig generated) code. These deprecations are thrown by the python interpreter at runtime. C++ and code generation aren't in play in this runtime scope. For deprecations that originate at the C/C++ level, that needs to be handled at compilation/code generation time. The solution there looks very different. 

Now, one caveat; if you write a python module then you could happily use the bskLogger for that purpose. But that's for general logging within the module, not system wide warnings like a deprecation.

Does this help, or perhaps I'm missing your point 🙃 ?

Reply:
Well, the use case scenario that I had in mind is: we create a new `SuperDragDynamicEffector` class that should be prefered over the old `DragDynamicEffector`. We still want to support `DragDynamicEffector` so that no user scripts break without warning. 

Ideally, we would like the following code to raise some kind of warning, but still work:
```
from Basilisk.simulation import dragEffector

dragEffector = dragEffector.DragDynamicEffector() # Should raise a warning: "Use the newer..."
```

I do not think this is something achievable at compile time. However, we could add a call in the constructor of `DragDynamicEffector` so that the warning is raised iff `DragDynamicEffector` is ever used in user scripts:

```
DragDynamicEffector::DragDynamicEffector()
{
    ...
    bskLogger.log(...); // or whatever other system
}
```

There are some python-only ways of achieveing something similar, such as through the use of the `pythonpreppend` SWIG features for deprecated classes or methods. However, these would only ever get called if the constructor or method was called from the Python layer directly; if a C++ function were to call a deprecated C++ method, any warning appended on the python layer through SWIG would be missed.

This is the use case that I was refering to. Maybe I am missing a way to do this or maybe there is need for such system. Thanks for your patience and detailed responses by the way, I really value your insight :)